src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -98,7 +98,7 @@ class MarathonScheduler @Inject() (
     log.info("Received status update for task %s: %s (%s)"
       .format(status.getTaskId.getValue, status.getState, status.getMessage))
 
-    val appId = taskIdUtil.appId(status.getTaskId)
+    val appId = taskIdUtil.appId(status.getTaskId).get
```

Comment:
This will throw an exception if the Try was a Failure.  It's _never_ ok to call `.get` on a `Try` unless you've first checked `.isSuccess`.  However in practice you should map the Try, or pattern-match to determine the result instead.

As an example of how to handle it in this case, it can combined with the following optional value roughly as follows:

```scala
val maybeAppId = taskIdUtil.appId(status.getTaskId)
val maybeTask = appId.toOption.flatMap { appId =>
  taskTracker.fetchTask(appId, status.getTaskId.getValue)
}
```

Or alternatively,

```scala
for (
  appId <- taskIdUtil.appId(status.getTaskId)
  marathonTask <- taskTracker.fetchTask(appId, status.getTaskId.getValue)
) {
  //...
```

Make sense?